src/driver-logitech-g300.c

```diff
@@ -517,9 +576,22 @@ logitech_g300_write_profile(struct ratbag_profile *profile)
 		raw_button->code = logitech_g300_button_action_to_raw(action);
 		raw_button->modifier = 0x00;
 		raw_button->key = 0x00;
-		if (action->type == RATBAG_BUTTON_ACTION_TYPE_KEY) {
+
+		if (action->type == RATBAG_BUTTON_ACTION_TYPE_MACRO) {
```

Comment:
You should probably check here that the macro is compatible and is not a 'long' macro.

Reply:
ratbag_action_keycode_from_macro will now return an error for a long macro.